deps/rabbit/include/mc.hrl

```diff
@@ -26,3 +12,30 @@
 -define(ANN_RECEIVED_AT_TIMESTAMP, rts).
 -define(ANN_DURABLE, d).
 -define(ANN_PRIORITY, p).
+
+%% RabbitMQ >= 3.13.3
+-type death_v2_anns() :: #{ttl => OriginalTtlHeader :: non_neg_integer()}.
+-record(death_v2, {source_queue :: rabbit_misc:resource_name(),
```

Comment:
The motivations for a new `death_v2` record were the following:
1. The old `deaths` record contains redundant fields `first` and `last`. These 2 fields are redundant because we already store `x-first-death-queue`, `x-first-death-reason`, `x-last-death-queue`, `x-last-death-reason` as mc annotations. Since the only remaining `deaths` field is `records`, we can simplify as done in this commit: The mc `deaths_v2` annotation will store directly a list of `death_v2` (instead of a `deaths` records containing a `records` field as done previously, which is just another level of indirection).
2. The on disk format of dead letter infos changes, as you suggested, from a map to a proplist. I find it easier to encode such changes somewhere as either a version number or into the record name (here: `deaths_v2`). Explicitly encoding the version will simplify debugging in the future. Also, new code can simply use the version number instead of checking whether a map or a proplist is used to determine whether v1 or v2 is in use as for example done in https://github.com/rabbitmq/rabbitmq-server/blob/4e5bfa422b559f7f15463fe3c48c2550bc694a90/deps/rabbit/src/mc_amqp.erl#L405

I pushed a commit to re-use the `deaths` record.

Reply:
Now seeing how the code looks like with reusing the `deaths` record, I prefer this reuse version. Thank you! :)